rhel/vex/parser_test.go

```diff
@@ -121,19 +181,54 @@ func TestWalkRelationships(t *testing.T) {
 			},
 			in:               "CROS:J-MOD:J-COMP:JMC",
```

Comment:
It's odd `J-COMP` isn't used, but I know you said you've only seen something like `CROS:J-MOD:JMC` in the wild, and having that fourth thing is rather unexpected, so the behavior this is testing is fine with me

Reply:
The test is to verify that no matter how many relationships we need to resolve we still pick/identify the correct components. Thinking about how many clair instances will be in the wild we wouldn't be able to react if the data changes (and it would be within the CSAF spec if prodsec created more complex relationships).